Finetune EVerest core build and fix/extend firmware update

mhei opened the pull request:
This PR bundles two points:
- it finetunes the chargebyte's default image build for Tarragon platform
- it fixes and extends config file handling during firmware update

mhei committed:
packagegroup-bsp: add dependency for full tar

The rauc update handlers in this layer assume that the full version
of tar is available which has some morre options thatn the busybox
variant.
So we need to add this dependency in this layer to pull it into
the image.

Signed-off-by: Michael Heimpold <michael.heimpold@chargebyte.com>

mhei committed:
core-bundlle: add update handler for EVerest config

This handler takes care of our default use-cases and assumptions
during firmware update process.

Signed-off-by: Michael Heimpold <michael.heimpold@chargebyte.com>

mhei committed:
everest-core: ensure that /var/lib/everest is present

This is a straight-forward way to include the desired path in the image.

Signed-off-by: Michael Heimpold <michael.heimpold@chargebyte.com>

mhei committed:
everest-core: drop docker stuff from image

Signed-off-by: Michael Heimpold <michael.heimpold@chargebyte.com>

mhei committed:
everest-core: build only desired modules

Strip undesired modules from the build to reduce the image footprint
and improve build time.

Signed-off-by: Michael Heimpold <michael.heimpold@chargebyte.com>

mhei committed:
core-bundle: fix config file migration

Former builds accidentially missed the full variant of tar and
the busybox's variant does not understand the --ignore-failed-read
parameter, so let's add a workaround.

Signed-off-by: Michael Heimpold <michael.heimpold@chargebyte.com>

barsnick commented on bundles/core-bundle/post-install.d/00-tar-helper:
```diff
@@ -0,0 +1,13 @@
+#!/bin/sh
+#
+# This script is a helper to be sourced in scripts which uses tar. It tries to detect which
```
```suggestion
# This script is a helper to be sourced in scripts which use tar. It tries to detect which
```

barsnick commented on bundles/core-bundle/post-install.d/00-tar-helper:
```diff
@@ -0,0 +1,13 @@
+# variant of tar is available to workaround passing the parameter --ignore-failed-read to
+# busybox's tar which does not know about this option.
+# For this the variable IGNORE_FAILED_READ is set to "--ignore-failed-read"
+# or empty which can then used in tar invocations.
```
```suggestion
# or empty which can then be used in tar invocations.
```

mhei committed:
core-bundlle: fix config.yaml replacement

Add a second condition to check whether a current EVerest configuration
file exists at all. This prevents replacing the config.yaml symlink
with an empty file for example when upgrading from proprietary
charging stack to EVerest-based one.

Signed-off-by: Michael Heimpold <michael.heimpold@chargebyte.com>

barsnick reviewed (approved):
Only spelling/grammar nitpicks, no priority to fix these.

mhei committed:
core-bundle: fix spelling/grammar nitpicks

Reported-by: Moritz Barsnick <moritz.barsnick@chargebyte.com>
Signed-off-by: Michael Heimpold <michael.heimpold@chargebyte.com>
